### PR TITLE
Update available Autoscaling metrics

### PR DESCRIPTION
**Pull request by david-wells-1**
Autoscaling groups can now provide additional metrics, this PR allows for these new metrics to be enabled.

Documentation is also updated.

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):
```
resource/aws_autoscaling_group: Allow enabling of all autoscaling group metrics
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSAutoScalingGroup_withMetrics'

==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSAutoScalingGroup_withMetrics -timeout 120m
=== RUN   TestAccAWSAutoScalingGroup_withMetrics
=== PAUSE TestAccAWSAutoScalingGroup_withMetrics
=== CONT  TestAccAWSAutoScalingGroup_withMetrics
--- PASS: TestAccAWSAutoScalingGroup_withMetrics (89.38s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	89.441s

...
```

### CODE REVIEW

File: aws/resource_aws_autoscaling_group_test.go [hunk 1]
```diff
@@ -2775,7 +2781,15 @@ resource "aws_autoscaling_group" "bar" {
...
-  	     "GroupMaxSize"
+  	     "GroupInServiceInstances",
+  	     "GroupMinSize",
+  	     "GroupMaxSize",
+  	     "GroupPendingCapacity",
+  	     "GroupInServiceCapacity",
+  	     "GroupStandbyCapacity",
+  	     "GroupTotalCapacity",
+  	     "GroupTerminatingCapacity",
+  	     "GroupStandbyInstances"
```
Comment (bflad) [inline aws/resource_aws_autoscaling_group_test.go]: The `TestAccAWSAutoScalingGroup_withMetrics` test is currently attempting to verify an API update of this argument and having the two lists be the same will not trigger the API update call. I recommend reverting this section of changes back.

File: aws/resource_aws_autoscaling_group_test.go [hunk 2]
```diff
@@ -2739,7 +2739,13 @@ resource "aws_autoscaling_group" "bar" {
   	     "GroupDesiredCapacity",
   	     "GroupInServiceInstances",
   	     "GroupMinSize",
-  	     "GroupMaxSize"
```
[the remark this answers is not in the document]
Comment (david-wells-1) [reply aws/resource_aws_autoscaling_group_test.go]: Thanks for the review

I have now updated the test - which passes

File: aws/resource_aws_autoscaling_group_test.go (hunk 1 above)
Comment (david-wells-1) [reply aws/resource_aws_autoscaling_group_test.go]: Thanks - this has now been reverted

**Comment by david-wells-1** [comment]
@bflad All requested amendments have now been updated. 

Output from `TestAccAWSAutoScalingGroup_withMetrics` also added.

**Pull request merged**